src/lib/message/L1ToL2MessageGasEstimator.ts

```diff
@@ -253,4 +256,84 @@ export class L1ToL2MessageGasEstimator {
       deposit,
     }
   }
+
+  /**
+   * Transactions that make an L1->L2 message need to estimate L2 gas parameters
+   * This function does that, and populates those parameters into a transaction request
+   * @param dataFunc
+   * @param l1Provider
+   * @param gasOverrides
+   * @returns
+   */
+  public async populateL1ToL2FunctionParams(
+    /**
+     * Function that will internally make an L1->L2 transaction
+     * Will initially be called with dummy values to trigger a special revert containing
+     * the real params. Then called again with the real params to form the final data to be submitted
+     */
+    dataFunc: (params: {
+      gasLimit: BigNumber
+      maxFeePerGas: BigNumber
+      maxSubmissionFee: BigNumber
```

Comment:
any reason not to use `L1ToL2MessageGasParams` as the type here?

Reply:
No, reused now